managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java

```diff
@@ -188,31 +178,22 @@ public Result retrieveKey(UUID customerUUID, UUID universeUUID) {
         ));
         byte[] keyRef = null;
         byte[] recoveredKey = null;
-        try {
-            ObjectNode formData = (ObjectNode) request().body().asJson();
-            keyRef = Base64.getDecoder().decode(formData.get("reference").asText());
-            UUID configUUID = UUID.fromString(formData.get("configUUID").asText());
-            recoveredKey = keyManager.getUniverseKey(universeUUID, configUUID, keyRef);
-            if (recoveredKey == null || recoveredKey.length == 0) {
-                final String errMsg = String.format(
-                        "No universe key found for universe %s",
-                        universeUUID.toString()
-                );
-                throw new RuntimeException(errMsg);
```

Comment:
This used to result in BAD_REQUEST. Now it will result in INTERNAL_SERVER_ERROR. To me looks like BAD_REQUEST is correct for this. No?

Reply:
Yeah BAD_REQUEST or NOT_FOUND makes sense. Will change it to BAD_REQUEST for now.